cmd/create/machinepool/nodepool.go

```diff
@@ -540,7 +543,27 @@ func getSubnetFromAvailabilityZone(cmd *cobra.Command, r *rosa.Runtime, isAvaila
 		}
 		r.Reporter.Infof("There are several subnets for availability zone '%s'", availabilityZone)
 		interactive.Enable()
-		subnet := getSubnetFromUser(cmd, r, false, cluster)
+
+		// should only get the subnets from the selected availability zone
+		var subnetOptions []string
+		for _, subnet := range privateSubnets {
```

Comment:
Is it required that we re-iterate through the list of privateSubnets to find which private subnets exist in a selected az zone?

Please correct me if I am wrong but wouldn't subnetsMap[availabilityZone] already have the same subnet ids (list of string), which line 551 is already trying to collect again?

Please kindly advise. :)

Reply:
@davidleerh Yep so previously, the subnetOptions was initiated within getSubnetFromUser. Unfortunately, subnetsMap[availabilityZone] only contains subnetid, which is different from subnetOptions where subnet has a type of types.Subnet. aws.SetSubnetOption(subnet)) needs types.Subnet to construct the string in proper option format. 

In line 550, I was checking if each subnet has the id that is listed from subnetsMap[availabilityZone]

Reply:
Sorry @chenz4027; 
Adding more eyes to this due to my lack of comprehension :cry: 

Hi @gdbranco would you kindly take a quick look at this? 
Might be my lack of understanding here but I feel like the value stored in subnetOptions == subnetsMap[availabilityZone] and that the iteration is extra compute that is not required?

subnetOptions is a list of strings which should be the same as datatype of subnetsMap[availabilityZone].

Reply:
I'd say the map can be expanded I suppose, to contain also the subnet type itself. I'm not sure how big this slice ends up being able to be, but I'd say saving some iterations is always worth

Reply:
Refactored, @gdbranco @davidleerh would you mind taking another look?

Reply:
lgtm